src/orquestra/sdk/_base/_dsl.py

```diff
@@ -389,27 +389,33 @@ class TaskDef(Generic[_P, _R], wrapt.ObjectProxy):
     def __init__(
         self,
         fn: Callable[_P, _R],
-        fn_ref: FunctionRef,
-        source_import: Import,
         output_metadata: TaskOutputMetadata,
+        source_import: Optional[Import] = None,
         parameters: Optional[OrderedDict] = None,
         dependency_imports: Optional[Tuple[Import, ...]] = None,
         resources: Resources = Resources(),
         custom_image: Optional[str] = None,
         custom_name: Optional[str] = None,
+        fn_ref: Optional[FunctionRef] = None,
     ):
         if isinstance(fn, BuiltinFunctionType):
             raise NotImplementedError("Built-in functions are not supported as Tasks")
         super(TaskDef, self).__init__(fn)
         self.__sdk_task_body = fn
-        self.fn_ref = fn_ref
-        self.source_import = source_import
+        self.fn_name = fn.__name__
         self.output_metadata = output_metadata
         self.parameters = parameters
-        self.dependency_imports = dependency_imports
         self.resources = resources
         self.custom_image = custom_image
         self.custom_name = custom_name
+        self.dependency_imports = dependency_imports
+        self._use_default_dependency_imports = dependency_imports is None
+        self.source_import = source_import
+        self._use_default_source_import = source_import is None
+        self.fn_ref = fn_ref
+
+        # task itself is not part of any workflow yet. Don't pass wf defaults
+        self.resolve_task_source_data()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏We spoke about this last week, not sure the resolution, should we make these `_` prefixed as private?

Reply:
true, forgot about that. Will take a look